youtube_dl/extractor/panopto.py

```diff
@@ -0,0 +1,277 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+from ..compat import compat_str
+
+from ..utils import (
+    ExtractorError,
+    smuggle_url,
+    unsmuggle_url,
+)
+
+import re
+from random import random
+import json
+
+
+class PanoptoBaseIE(InfoExtractor):
+    """The base class with common methods for Panopto extractors."""
+
+    @classmethod
+    def _match_organization(cls, url):
+        """Match and return the organization part of a Panopto hosted URL."""
+        if '_VALID_URL_RE' not in cls.__dict__:
+            cls._VALID_URL_RE = re.compile(cls._VALID_URL)
+        m = cls._VALID_URL_RE.match(url)
+        assert m
+        return compat_str(m.group('org'))
+
+
+class PanoptoIE(PanoptoBaseIE):
+    """Extracts a single Panopto video including all available streams."""
+
+    _VALID_URL = r'^https?://(?P<org>[a-z0-9]+)\.hosted\.panopto.com/Panopto/Pages/Viewer\.aspx\?id=(?P<id>[a-f0-9-]+)'
+    _TESTS = [
+        {
+            'url': 'https://demo.hosted.panopto.com/Panopto/Pages/Viewer.aspx?id=26b3ae9e-4a48-4dcc-96ba-0befba08a0fb',
+            'md5': 'e8e6ef6b0572dd5985f5f8c3e096f717',
```

Comment:
```suggestion
            'md5': '048dab5c2f8ab97f2bd75ab4cf3f463a',
```
Maybe this video was changed, this now seems to be the correct hash.